terasoluna-tourreservation-web/src/main/java/org/terasoluna/tourreservation/app/managereservation/ReservationReportPdfView.java

```diff
@@ -0,0 +1,115 @@
+        form.setField("childCount", String.valueOf(downloadPDFOutput
+                .getChildCount()));
+        form.setField("tourName", downloadPDFOutput.getTourName());
+        form.setField("accomName", downloadPDFOutput.getAccomName());
+        form.setField("customerKana", downloadPDFOutput.getCustomerKana());
+        form.setField("customerTel", downloadPDFOutput.getCustomerTel());
+        form.setField("adultUnitPrice", String.valueOf(downloadPDFOutput
+                .getAdultUnitPrice()));
+        form.setField("reservedDay", String.valueOf(sdf.format(downloadPDFOutput
+                .getReservedDay())));
+        form.setField("conductor", downloadPDFOutput.getConductor());
+        form.setField("tourAbs", downloadPDFOutput.getTourAbs());
+        form.setField("customerAdd", downloadPDFOutput.getCustomerAdd());
+        form.setField("customerJob", downloadPDFOutput.getCustomerJob());
+        form.setField("tourDays", downloadPDFOutput.getTourDays());
+        form.setField("depDay", String.valueOf(sdf.format(downloadPDFOutput
+                .getDepDay())));
+        form.setField("customerName", downloadPDFOutput.getCustomerName());
+        form.setField("childUnitPrice", String.valueOf(downloadPDFOutput
+                .getChildUnitPrice()));
+        form.setField("depName", downloadPDFOutput.getDepName());
+        form.setField("customerBirth", String.valueOf(sdf.format(
+                downloadPDFOutput.getCustomerBirth())));
+        form.setField("arrName", downloadPDFOutput.getArrName());
+        form.setField("customerMail", downloadPDFOutput.getCustomerMail());
+        form.setField("adultCount", String.valueOf(downloadPDFOutput
+                .getAdultCount()));
+        form.setField("customerCode", downloadPDFOutput.getCustomerCode());
+        form.setField("reserveNo", downloadPDFOutput.getReserveNo());
+        form.setField("remarks", downloadPDFOutput.getRemarks());
+        form.setField("accomTel", downloadPDFOutput.getAccomTel());
+        form.setField("customerPost", downloadPDFOutput.getCustomerPost());
+        form.setField("printDay", String.valueOf(sdf.format(downloadPDFOutput
+                .getPrintDay())));
+        form.setField("adultPrice", String.valueOf(downloadPDFOutput
+                .getAdultPrice()));
+        form.setField("childPrice", String.valueOf(downloadPDFOutput
+                .getChildPrice()));
+        form.setField("sumPrice", String.valueOf(downloadPDFOutput
+                .getSumPrice()));
+        form.setField("paymentTimeLimit", downloadPDFOutput
+                .getPaymentTimeLimit());
+        stamper.setFormFlattening(true);
+        stamper.setFreeTextFlattening(true);
+        stamper.close();
+
+        response.setContentType("application/pdf");
+        response.setCharacterEncoding("UTF-8");
+        response.setHeader("Content-Disposition", "attachment; filename="
+                + downloadPDFOutput.getPdfName());
```

Comment:
- [ ] Recommend to set the header and encoding on overrided `AbstractView#prepareResponse` or `AbstractView#writeToResponse`.
Because `AbstractPdfStamperView#mergePdfDocument` sounds the method only merging Pdf and model.

Reply:
6f17ff5421d497c352f72270d293b06918ef1b23

I changed to set header and encoding in overrided `AbstractView#prepareResponse` in above commit .

Reply:
You should call `super#prepareResponse`. You shouldn't re-implement super class method.

```java
@Override
protected void prepareResponse(...) {
  super.prepareResponse(...);
  response.setCharacterEncoding("UTF-8");
  response.setHeader("Content-Disposition", ...);
}
```

Reply:
e2bfa3c4c1a13895e2427b0d8e21e5eed67d6120

Finally, I decided to set the PDF name during the method of `mergePDFDocument`, not overriding`AbstractView#prepareResponse`. (I followed [the idea that PDF names are part of PDF](https://github.com/terasolunaorg/terasoluna-tourreservation/pull/608#discussion_r260198798). )

Reply:
NG. Use `AbstractView#prepareResponse` to set `characterEncoding`.

Reply:
3b9e59e4edf17ef3673a7733eda7ee913d678e0d

I changed to set `characterEncoding` in `AbstractView#prepareResponse` .